go/ai/prompt.go

```diff
@@ -426,3 +430,174 @@ func renderDotprompt(templateText string, variables map[string]any, defaultInput
 	}
 	return str, nil
 }
+
+// LoadPromptFolder loads prompts and partials from the input directory for the given namespace
+func LoadPromptFolder(r *registry.Registry, dir string, namespace string) error {
+	useDefaultDir := false
+	if dir == "" {
+		dir = "./prompts"
+		useDefaultDir = true
+	}
+	promptsPath, err := filepath.Abs(dir)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to resolve directory: %v", err)
+		if !useDefaultDir {
+			return errors.New(errMsg)
+		}
+		slog.Error(errMsg)
+		return nil
+	}
+	if _, err := os.Stat(promptsPath); os.IsNotExist(err) {
+		errMsg := fmt.Sprintf("Prompts directory does not exist: %s", promptsPath)
+		if !useDefaultDir {
+			return errors.New(errMsg)
+		}
+		slog.Error(errMsg)
+		return nil
+	}
+	return LoadPromptFolderRecursively(r, promptsPath, namespace, "")
+}
+
+// LoadPromptFolderRecursively recursively loads prompts and partials from the directory.
+func LoadPromptFolderRecursively(r *registry.Registry, dir string, namespace string, subDir string) error {
+	parentPath := filepath.Join(dir, subDir)
+	entries, err := os.ReadDir(parentPath)
+	if err != nil {
+		slog.Error("Failed to read directory", "error", err.Error())
+		return nil
+	}
+
+	for _, entry := range entries {
+		fileName := entry.Name()
+		if entry.IsDir() {
+			err := LoadPromptFolderRecursively(r, dir, namespace, filepath.Join(subDir, fileName))
+			if err != nil {
+				return err
+			}
+		} else if strings.HasSuffix(fileName, ".prompt") {
+			if strings.HasPrefix(fileName, "_") {
+				partialName := strings.TrimSuffix(fileName[1:], ".prompt")
+				source, err := os.ReadFile(filepath.Join(parentPath, fileName))
+				if err != nil {
+					slog.Error("Failed to read partial file", "error", err.Error())
+					continue
+				}
+				definePartial(r, partialName, string(source))
+				slog.Debug("Registered Dotprompt partial \"%s\" from \"%s\"", partialName, filepath.Join(parentPath, fileName))
+			} else {
+				err := loadPrompt(r, dir, fileName, subDir, namespace)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// definePartial registers a partial template in the registry.
+func definePartial(r *registry.Registry, name string, source string) {
+	// TODO: Add this functionality
+}
+
+// loadPrompt loads a single prompt into the registry.
+func loadPrompt(r *registry.Registry, path string, filename string, prefix string, namespace string) error {
+	name := strings.TrimSuffix(filename, ".prompt")
+	var variant string
+	if strings.Contains(name, ".") {
+		parts := strings.SplitN(name, ".", 2)
+		name = parts[0]
+		variant = parts[1]
```

Comment:
```suggestion
	name, variant, _ := strings.Cut(name, ".")
```

I think this will do the same, please verify.